app/services/ReservationService.java

```diff
@@ -0,0 +1,191 @@
+package services;
+
+import models.*;
+import play.Logger;
+import repository.ReservationRepository;
+import repository.RestaurantRepository;
+import repository.TableRepository;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Created by lapa on 5/25/17.
+ */
+@Singleton
+public class ReservationService implements BaseService {
+
+    private final Long HALF_HOUR = 1800000L;
+    private final Long HOUR = 3600000L;
+    private final Long RESERVATION_PERIOD = 7200000L;
+    private Long FIRST_RESERVATION = null;
+    private Long LAST_RESERVATION = null;
+
+
+    private ReservationRepository reservationRepository;
+    private RestaurantRepository restaurantRepository;
```

Comment:
Service classes should interact with other services, not repositories (only `ReservationRepository ` should be injected in `ReservationService `, other classes should be services).